simulator/src/main/java/com/hedera/block/simulator/grpc/impl/PublishStreamServerObserver.java

```diff
@@ -0,0 +1,121 @@
+import com.hedera.hapi.block.protoc.PublishStreamRequest;
+import com.hedera.hapi.block.protoc.PublishStreamResponse;
+import com.hedera.hapi.block.protoc.PublishStreamResponse.Acknowledgement;
+import com.hedera.hapi.block.protoc.PublishStreamResponse.BlockAcknowledgement;
+import com.hedera.hapi.block.stream.protoc.BlockItem;
+import com.hedera.hapi.block.stream.protoc.BlockProof;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+import java.util.Deque;
+import java.util.List;
+
+/**
+ * A gRPC stream observer implementation that handles incoming {@link PublishStreamRequest} messages
+ * on the server side. This observer processes incoming block stream publications, maintains stream status
+ * history, and manages the response stream back to the client. It implements flow control by tracking
+ * stream state and enforcing capacity limits on status history.
+ */
+public class PublishStreamServerObserver implements StreamObserver<PublishStreamRequest> {
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+
+    // Service dependencies
+    private final MetricsService metricsService;
+
+    // gRPC Components
+    private final StreamObserver<PublishStreamResponse> responseObserver;
+
+    // State
+    private final int lastKnownStatusesCapacity;
+    private final Deque<String> lastKnownStatuses;
+
+    /**
+     * Constructs a new PublishStreamServerObserver that handles stream requests and maintains a history of statuses.
+     *
+     * @param responseObserver The observer that handles responses back to the client
+     * @param metricsService The service for recording consumption metrics
+     * @param lastKnownStatuses A deque to store the history of request statuses
+     * @param lastKnownStatusesCapacity The maximum number of statuses to maintain in the history
+     * @throws NullPointerException if responseObserver or lastKnownStatuses is null
+     */
+    public PublishStreamServerObserver(
+            @NonNull final StreamObserver<PublishStreamResponse> responseObserver,
+            @NonNull final MetricsService metricsService,
+            @NonNull final Deque<String> lastKnownStatuses,
+            final int lastKnownStatusesCapacity) {
+        this.responseObserver = requireNonNull(responseObserver);
+        this.metricsService = requireNonNull(metricsService);
+        this.lastKnownStatuses = requireNonNull(lastKnownStatuses);
+        this.lastKnownStatusesCapacity = lastKnownStatusesCapacity;
+    }
+
+    /**
+     * Processes incoming publish stream requests, maintaining a history of requests and handling block acknowledgements.
+     * If the request contains block items with a block proof, generates and sends an acknowledgement response.
+     *
+     * @param publishStreamRequest The incoming stream request to process
+     */
+    @Override
+    public void onNext(PublishStreamRequest publishStreamRequest) {
```

Comment:
In future PRs we will add handlers here to respond, depending on predefined configuration or depending on controller (don't exist yet). For now we have only one handler, which returns block ack.